pkg/porter/generateManifest.go

```diff
@@ -49,5 +58,64 @@ func (p *Porter) generateInternalManifest(opts BuildOptions) error {
 		}
 	}
 
+	// find all referenced images that does not have digest specified
+	// get the image digest for all of them and update the manifest with the digest
+	err = e.WalkNodes(ctx, "images.*", func(ctx context.Context, nc *yqlib.NodeContext) error {
```

Comment:
I modified it. Hope it will make it more clear.
I left it as an anonymous function is has couple reasons: 
- from my perspective, this function is very specific to `WalkNodes` since we expect the `yqlib.NodeContext` is passed in as an argument. I think it helps for readability to show that the NodeContext here is specific to the nodes that we are traversing.
- in this anonymous function, we need access to the editor to update the yaml file once we find the correct node. I would like to keep the callback function to only have access to the node. Any side affect caused by the callback function should be managed by the caller.